### ISSUE TITLE
GUI smart and custom backup retention aren't covered

### ISSUE THREAD

**Issue by ts678** [kees-z/DuplicatiDocs#83]
[Creating a new backup job](https://duplicati.readthedocs.io/en/latest/03-using-the-graphical-user-interface/#creating-a-new-backup-job)

> The retention can be set in 3 ways

should cover `Smart backup retention` and `Custom backup retention`. [Here](https://github.com/duplicati/duplicati/blob/b9aad24deb4a4c88bb284bce3a6ee64e1939ce9e/Duplicati/Server/webroot/ngax/scripts/controllers/EditBackupController.js#L3-L3) one sees smart is  just a reasonable preconfigure of `1W:1D,4W:1W,12M:1M` which custom can vary from. The help text shows this only by description. Not sure what manual should do.

I'm also not decided if hours minutes and seconds (h m s) should be mentioned or are too obscure. One time that hours is useful  is when someone likes having one backup per day kept, e.g. they use `Smart backup retention`, and they backup once per day. Any timing variations may delete one that follows the previous by, say, 23 hours and 59 minutes (or worse if they run manual backups). This "skip" winds up looking like one every other day, so is pretty obvious and a bit unsettling, and invites forum topics asking why.

Some people also take terminology like days and weeks as a calendar, but it's not. All larger terms wind up converting into seconds. This is especially obvious in the paragraph above, where a common backup frequency (daily) bumps into a sensible retention policy (daily). It seems that complaints about years, months, and weeks not following a calendar are rare to none. How to document time?

Occasionally some people don't quite understand how time frames work, so try to make them one after another instead of nesting. Sometimes I've tried to use the term "age" to throw in a common term. A diagram (shaded for backup "density"?) might be helpful.

Another thing whose documentation varies even in help is `U`. Command line help says it can "indicate an unlimited time interval", which is confusing because there were two concepts "time frame" and "interval" defined, so which one takes a "U"? I think both do.

From a time frame perspective, one might use it to mean "unlimited time frame" which implicitly means "beyond the largest specific time" because time frames are sorted. This also prevents implied delete beyond the largest time frame (see topic below). For interval, there is no minimum interval. I sometimes use language like "spacing" and "thinning" in forum posts,, but this is all tough to explain completely. I suspect lots of people just see "Smart" in the GUI and pick it without much thought, which is actually reasonable to do.

People who actually think this through thoroughly may realize (or may be told in the forum) that a file which exists (between creation and deletion) for a time less than current minimum interval (which can increase with time) may disappear completely. This may matter to some people, but there's a limited amount that can be said. Would it be better to have an "advanced" section for those wanting it?

[Removing files from backup](https://forum.duplicati.com/t/removing-files-from-backup/13014) is a recent forum topic pointing to big  2017 topic that predated the early 2018 changes. Links provided, and there are a lot more. It would be nice to get this in the manual. If need be, maybe staff can discuss a draft? Is markdown similar?

Whatever is said probably deserves some verification testing, and I suspect some coded help text might also merit an upgrade if a good way to explain retention policy can be determined. Maybe the manual can take the lead, then let that guide the other spots.

**Comment by runmode** [comment]
Hello, ts678 helped me to understand the retention policy, which I did not just when reading all the docs and many forum discussions. To consolidate the learnings I worded a [brief description of how Duplicati works](https://forum.duplicati.com/t/new-retention-policy-deletes-old-backups-in-a-smart-way/2195/135?u=runmode), which could help other users to get into Duplicati faster. It's an external view which might help externals...

In the forum certain questions come up again and again, and much time is spent with individual answers, for several reasons (find proposals for quick actions in brackets):

- People don't find a compact introduction in a context, rather must puzzle pieces of information together (feel free to use my text).
- Central terms used stand-alone, such as 'backup' and 'version', can lead to false interpretations by users who have the traditional backup concept in mind, e.g. "new backup version" or "purging a version" is missleading (instead index version, backup file, etc. should be used).
- The articles on the landing page https://www.duplicati.com/articles/ dated 2016, are outdated after the online user manual 2023 which seems to be an updated version (as you have the online manual, the articles could be removed)
- In the forum hundreds of old discussions with outdated content are still available, which is rather missleading than helpful (if people find answers in the user manual they don't need to search in the forum).

In reply to your post: If the user manual simply informed how to use the time frames and the 'U', no discussion is required how users tend to interpret these options (and if the code should follow some user's guessing). So if someone writes in the docs that the time frames overlap by design, you don't need to change that. If people feel unsave with the thinning between intervals, just explain how it works. 

My one-page description of 'How Duplicati works' covers most questions, I hope it can be of use. All the best!
